src/main/java/com/github/tomakehurst/wiremock/stubbing/AbstractStubMappings.java

```diff
@@ -66,183 +73,203 @@ public AbstractStubMappings(
     this.v2transformers = v2transformers;
     this.filesFileSource = new BlobStoreFileSource(filesBlobStore);
     this.stubLifecycleListeners = stubLifecycleListeners;
+    this.failIfMultipleMappingsMatch = failIfMultipleMappingsMatch;
   }
 
   @Override
-  public ServeEvent serveFor(ServeEvent initialServeEvent) {
+  public ServeEvent serveFor(final ServeEvent initialServeEvent) {
     final LoggedRequest request = initialServeEvent.getRequest();
 
     final List<SubEvent> subEvents = new LinkedList<>();
 
-    StubMapping matchingMapping =
-        store
-            .findAllMatchingRequest(request, customMatchers, subEvents::add)
-            .filter(
-                stubMapping ->
-                    stubMapping.isIndependentOfScenarioState()
-                        || scenarios.mappingMatchesScenarioState(stubMapping))
-            .findFirst()
-            .orElse(StubMapping.NOT_CONFIGURED);
+    final StubMapping matchingMapping = this.getMatchingMapping(request, subEvents);
 
     subEvents.forEach(initialServeEvent::appendSubEvent);
 
-    scenarios.onStubServed(matchingMapping);
+    this.scenarios.onStubServed(matchingMapping);
```

Comment:
A bunch of unrelated refactorings, but not blocking the PR because of it. I see no strict need to use `this.` in the code or in the code style though